old/lib/LedgerSMB/DBObject/User.pm

```diff
@@ -101,22 +102,24 @@ sub get_option_data {
     }
     use warnings;
 
-    foreach my $key ( all_country_codes() )
-    {
-        push @{$self->{country_codes}}, {
-            label => code2country($key),
-            id => $key,
-        };
-    }
-    @{$self->{country_codes}} =
-        sort { $a->{label} cmp $b->{label} } @{$self->{country_codes}};
+    # Pull supported languages codes
+    my @rows = $self->call_dbmethod(funcname => 'person__list_languages');
+
+    my %regions = %{$locale->all_regions}; # Localized countries
+    my %languages = %{$locale->all_languages()}; # Localized languages
 
-    foreach my $key ( all_language_codes() )
-    {
+    # Localize the list, making sure to keep the language key as is
+    foreach my $row ( @rows ) {
+        my ($language,$region) = split /_/, $row->{code};
+        # Use the language_country localized version if available
```

Comment:
Maybe you can add a bit more explanation of what codes come from which sources and how you need to compensate each code to be able to compare them, with examples? That explains why the code does what it does.

Reply:
Done